upup/pkg/fi/vfs_castore.go

```diff
@@ -300,19 +302,11 @@ func (c *VFSCAStore) FindKeypair(id string) (*Certificate, *PrivateKey, error) {
 func (c *VFSCAStore) FindCert(id string) (*Certificate, error) {
 	var certs *certificates
 
-	if id == CertificateId_CA {
-		caCertificates, _, err := c.readCAKeypairs()
```

Comment:
So `readCAKeypairs` performs caching of the CA keypair, but it also is a little naughty and will generate the CA certificate if needed.  I guess though that this is required so that this can be used from nodeup, where we don't have access to the certificates.  I believe that the auto-generation of the CA certificate was a remnant of the template-driven generation, so should not be needed any more.  If we do need caching or autogeneration I think we can probably add caching back in, or we can explicitly generate the CA key as one of our first steps.